### PR TITLE
[BEAM-11205] Update Libraries BOM dependencies to version 25.2.0

### PR DESCRIPTION
**Pull request by benWize**
Updating Google Cloud Libraries BOM to the latest version 25.2.0 and associated dependencies to the compatible versions.

Updating dependencies listed in https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/25.2.0/artifact_details.html 

Linkage check results
https://gist.github.com/benWize/37f0d34e157974f6c1ff5a8e53f94a9b

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by benWize** [comment]
Run PythonLint PreCommit

**Comment by benWize** [comment]
Run Java PostCommit

**Comment by benWize** [comment]
Run PostCommit_Java_Hadoop_Versions

**Comment by benWize** [comment]
Run Dataflow ValidatesRunner

**Comment by benWize** [comment]
Run Java PostCommit

**Comment by benWize** [comment]
Run PostCommit_Java_Hadoop_Versions

**Comment by benWize** [comment]
Run Dataflow ValidatesRunner

**Comment by benWize** [comment]
Run Java PreCommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -575,11 +575,11 @@ class BeamModulePlugin implements Plugin<Project> {
         google_cloud_datastore_v1_proto_client      : "com.google.cloud.datastore:datastore-v1-proto-client:2.1.3",
         google_cloud_firestore                      : "com.google.cloud:google-cloud-firestore", // google_cloud_platform_libraries_bom sets version
         google_cloud_pubsub                         : "com.google.cloud:google-cloud-pubsub", // google_cloud_platform_libraries_bom sets version
-        google_cloud_pubsublite                     : "com.google.cloud:google-cloud-pubsublite:1.5.0",  // TODO(dpcollins-google): Let google_cloud_platform_libraries_bom set version once high enough
+        google_cloud_pubsublite                     : "com.google.cloud:google-cloud-pubsublite",  // google_cloud_platform_libraries_bom sets version
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:578]: Nice.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -455,16 +455,16 @@ class BeamModulePlugin implements Plugin<Project> {
     def classgraph_version = "4.8.104"
     def errorprone_version = "2.10.0"
     // Try to keep gax_version consistent with gax-grpc version in google_cloud_platform_libraries_bom
-    def gax_version = "2.8.1"
+    def gax_version = "2.16.0"
     def google_clients_version = "1.32.1"
     def google_cloud_bigdataoss_version = "2.2.4"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:460]: Can you make it 2.2.6?

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 3]
```diff
@@ -455,16 +455,16 @@ class BeamModulePlugin implements Plugin<Project> {
...
     // Try to keep google_cloud_spanner_version consistent with google_cloud_spanner_bom in google_cloud_platform_libraries_bom
-    def google_cloud_spanner_version = "6.20.0"
-    def google_code_gson_version = "2.8.9"
+    def google_cloud_spanner_version = "6.23.3"
+    def google_code_gson_version = "2.9.0"
     def google_oauth_clients_version = "1.32.1"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:464]: Can you update this to 1.33.3

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 4]
```diff
@@ -455,16 +455,16 @@ class BeamModulePlugin implements Plugin<Project> {
...
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
     def grpc_version = "1.44.0"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:466]: Can you update this to 1.45.1?

**Review by suztomo — COMMENTED** [review]
Can you also update datastore-v1-proto-client to 2.2.10?

**Review by suztomo — CHANGES_REQUESTED** [review]
@benWize Can you set datastore-v1-proto-client to 2.2.10 and override proto-google-cloud-datastore-v1 to 0.93.10 (currently no explicit version)? It's consistent with datastore-v1-proto-client to 2.2.10.

C.f. https://search.maven.org/artifact/com.google.cloud/google-cloud-datastore-bom/2.2.10/pom

CC: @kolea2

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 4 above)
Comment (benWize) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Yes, updated

**Comment by benWize** [comment]
Run SQL PreCommit

**Comment by benWize** [comment]
Run Python_PVR_Flink PreCommit

**Comment by benWize** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 5]
```diff
@@ -661,10 +661,10 @@ class BeamModulePlugin implements Plugin<Project> {
         proto_google_cloud_bigtable_admin_v2        : "com.google.api.grpc:proto-google-cloud-bigtable-admin-v2", // google_cloud_platform_libraries_bom sets version
         proto_google_cloud_bigtable_v2              : "com.google.api.grpc:proto-google-cloud-bigtable-v2", // google_cloud_platform_libraries_bom sets version
         proto_google_cloud_datacatalog_v1beta1      : "com.google.api.grpc:proto-google-cloud-datacatalog-v1beta1", // google_cloud_platform_libraries_bom sets version
-        proto_google_cloud_datastore_v1             : "com.google.api.grpc:proto-google-cloud-datastore-v1", // google_cloud_platform_libraries_bom sets version
+        proto_google_cloud_datastore_v1             : "com.google.api.grpc:proto-google-cloud-datastore-v1:0.93.10", // google_cloud_platform_libraries_bom sets version
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:664]: My memo: This is in-line with com.google.cloud.datastore:datastore-v1-proto-client:2.2.10

(no action required)

**Comment by suztomo** [comment]
The change looks good to me.

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by suztomo** [comment]
Run PostCommit_Java_DataflowV2

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Where are Jenkins builds...?

**Comment by suztomo** [comment]
Run Java Precommit

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run Java Precommit

**Comment by benWize** [comment]
> org.apache.beam.examples.subprocess.ExampleEchoPipelineTest.testExampleEchoPipeline

There is nothing pending from my perspective.

**Comment by suztomo** [comment]
Nice. Let's merge when checks become green. I see "Dataflow Java Post Commit Tests" pending (running).

**Comment by benWize** [comment]
> Nice. Let's merge when checks become green. I see "Dataflow Java Post Commit Tests" pending (running).

All checks are passing now! Thanks, @suztomo

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
@benWize Thank you.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 5 above)
Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:664]: @ansh0l For upcoming change by you, remove this version part and let google_cloud_platform_libraries_bom set the version.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 6]
```diff
@@ -455,16 +455,16 @@ class BeamModulePlugin implements Plugin<Project> {
...
-    def google_code_gson_version = "2.8.9"
-    def google_oauth_clients_version = "1.32.1"
+    def google_cloud_spanner_version = "6.23.3"
+    def google_code_gson_version = "2.9.0"
+    def google_oauth_clients_version = "1.33.3"
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
-    def grpc_version = "1.44.0"
-    def guava_version = "31.0.1-jre"
+    def grpc_version = "1.45.1"
+    def guava_version = "31.1-jre"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:467]: @ansh0l  For the upcoming change by you, reference this
https://github.com/googleapis/java-shared-dependencies/blob/v3.0.1/first-party-dependencies/pom.xml#L58

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 7]
```diff
@@ -479,7 +479,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def postgres_version = "42.2.16"
     def powermock_version = "2.0.9"
     // Try to keep protobuf_version consistent with the protobuf version in google_cloud_platform_libraries_bom
-    def protobuf_version = "3.19.3"
+    def protobuf_version = "3.19.4"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:482]: @ansh0l For the upcoming change by you, reference this:
https://github.com/googleapis/java-shared-dependencies/blob/v3.0.1/first-party-dependencies/pom.xml#L58

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 6 above)
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:467]: @ansh0l  For the upcoming change by you, reference this
https://github.com/googleapis/java-shared-dependencies/blob/v3.0.1/first-party-dependencies/pom.xml#L58

Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:467]: Should the `google_code_gson_version` be changed here? The above pom file doesn't have any reference to this dependency.

CC: @kileys

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 5 above)
Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:664]: This change suggests changes in the version for `proto_google_cloud_datastore_v1`. Should we let the Datastore team make this change, and not us (Spanner team)?

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 6 above)
Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:467]: https://github.com/googleapis/java-shared-dependencies/blob/v3.0.1/third-party-dependencies/pom.xml has the gson version.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 5 above)
Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:664]: Can you (spanner team) take care of this?

Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:664]: Sure, we'll take care of this.

**Pull request merged**